llama-index-integrations/embeddings/llama-index-embeddings-azure-inference/llama_index/embeddings/azure_inference/base.py

```diff
@@ -0,0 +1,164 @@
+
+from azure.ai.inference import EmbeddingsClient
+from azure.ai.inference.aio import EmbeddingsClient as EmbeddingsClientAsync
+from azure.core.credentials import AzureKeyCredential
+
+DEFAULT_AZURE_INFERENCE_ENDPOINT = "https://models.inference.ai.azure.com"
+
+
+class AzureAIEmbeddingsModel(BaseEmbedding):
+    """Azure AI model inference for embeddings.
+
+    Examples:
+        ```python
+        from llama_index.core import Settings
+        from llama_index.embeddings.azure_inference import AzureAIEmbeddingsModel
+
+        llm = AzureAIEmbeddingsModel(
+            endpoint="https://[your-endpoint].inference.ai.azure.com",
+            credential="your-api-key",
+        )
+
+        # If using Microsoft Entra ID authentication, you can create the
+        # client as follows
+        #
+        # from azure.identity import DefaultAzureCredential
+        #
+        # embed_model = AzureAIEmbeddingsModel(
+        #     endpoint="https://[your-endpoint].inference.ai.azure.com",
+        #     credential=DefaultAzureCredential()
+        # )
+
+        # Once the client is instantiated, you can set the context to use the model
+        Settings.embed_model = embed_model
+
+        documents = SimpleDirectoryReader("./data").load_data()
+        index = VectorStoreIndex.from_documents(documents)
+        ```
+    """
+
+    model_extras: Dict[str, Any] = Field(
+        default_factory=dict, description="Additional kwargs model parameters."
+    )
+
+    _client: EmbeddingsClient = PrivateAttr()
+    _async_client: EmbeddingsClientAsync = PrivateAttr()
+
+    def __init__(
+        self,
+        endpoint: str = None,
+        credential: Union[str, AzureKeyCredential, "TokenCredential"] = None,
+        model_name: str = None,
+        embed_batch_size: int = DEFAULT_EMBED_BATCH_SIZE,
+        callback_manager: Optional[CallbackManager] = None,
+        num_workers: Optional[int] = None,
+        client_kwargs: Optional[Dict[str, Any]] = None,
+        **kwargs: Any
+    ):
+        client_kwargs = client_kwargs or {}
+
+        endpoint = get_from_param_or_env(
+            "endpoint",
+            endpoint,
+            "AZURE_INFERENCE_ENDPOINT_URL",
+            DEFAULT_AZURE_INFERENCE_ENDPOINT,
+        )
+        credential = get_from_param_or_env(
+            "credential", credential, "AZURE_INFERENCE_ENDPOINT_CREDENTIAL", None
+        )
+        credential = (
+            AzureKeyCredential(credential)
+            if isinstance(credential, str)
+            else credential
+        )
+
+        if not credential:
+            raise ValueError(
+                "You must provide an credential to use the Azure AI model inference LLM."
+            )
+        self._client = EmbeddingsClient(
+            endpoint=endpoint,
+            credential=credential,
+            user_agent="llamaindex",
+            **client_kwargs,
+        )
+
+        self._async_client = EmbeddingsClientAsync(
```

Comment:
Async clients with sync token credentials are problematic (it will make sync/blocking calls...)